presto-main/src/main/java/com/facebook/presto/server/WorkerResource.java

```diff
@@ -66,6 +69,44 @@ public Response getThreads(@PathParam("nodeId") String nodeId)
         return proxyJsonResponse(nodeId, "v1/thread");
     }
 
+    @GET
+    public Response getWorkerList()
+    {
+        Set<Node> nodes = nodeManager.getAllNodes().getActiveNodes();
+        Set<JsonNodeInfo> jsonNodes = new HashSet<>();
+        for (Node node : nodes) {
```

Comment:
Change this to use streams and collectors as per https://github.com/prestodb/presto#code-style